vignettes/04-synthesis-data.Rmd

```diff
@@ -1 +1,294 @@
+     'fullfield_L1_ua-mac_2018-05-05_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-06_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-08_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-09_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-10_rgb_stereovis_ir_sensors_fullfield_sorghum6_sun_may2018_-_copy_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-12_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-13_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-14_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-15_rgb_stereovis_ir_sensors_fullfield_sorghum6_sun_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-17_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-18_rgb_stereovis_ir_sensors_fullfield_sorghum6_sun_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-19_rgb_stereovis_ir_sensors_plots_sorghum6_sun_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-20_rgb_stereovis_ir_sensors_plots_sorghum6_shade_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-21_rgb_stereovis_ir_sensors_fullfield_sorghum6_shade_may2018_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-22_rgb_stereovis_ir_sensors_plots_sorghum6_sun_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-23_rgb_stereovis_ir_sensors_plots_sorghum6_sun_thumb.tif',
+     'fullfield_L1_ua-mac_2018-05-28_rgb_stereovis_ir_sensors_plots_sorghum6_shade_rgb_eastedge_mn_thumb.tif'
+     )
+```
+
+We will loop through these images, extract our plot data, and calculate the "greeness" of each extract.
+We are using the name of the file to extract the date for later.
+
+```{r synth_get_greeness}
+library(raster)
+library(stringr)
+
+# Extract the date from the file name
+getDate <- function(file_name){
+  date <- str_match_all(file_name, '[0-9]{4}-[0-9]{2}-[0-9]{2}')[[1]][,1]
+  return(date)
+}
+
+
+# Get the clip coordinates into the correct order
+clip_coords <- paste(toString(bounding_box[4]),' ',toString(bounding_box[3]),
+                    ' ',toString(bounding_box[2]),' ',toString(bounding_box[1]))
+
+
+# Returns the greeness value of the clipped image
+getGreeness <- function(file_name, clip_coords){
+  out_file <- "extract.tif"
+
+  # Execute the GDAL command to extract the plot  
+  command = paste("gdal_translate -projwin ",clip_coords," ",file_name," ",out_file)
+  system(command)
+
+  # Load the green band of the image and calculate the greeness value
+  green_image   <- raster(out_file, band = 2)
+  greeness      <- cellStats(green_image, stat = "mean")
```

Comment:
Wasn't this supposed to be some summary metric that used both the green and red bands? Or did yall decide to change it to something simpler?

Reply:
I think Chris may have missed discussion of the index ? Greenness alone is likely insufficient especially given the high variance in brightness across and even within images.